tests/EventTest.php

```diff
@@ -55,44 +37,26 @@ final class EventTest extends TestCase
 
     protected function setUp(): void
     {
-        $this->uuidGeneratorInvokationCount = 0;
-        $this->originalUuidFactory = new UuidFactory();
         $this->client = ClientBuilder::create()->getClient();
         $this->options = $this->client->getOptions();
-
-        /** @var UuidFactoryInterface|MockObject $uuidFactory */
-        $uuidFactory = $this->getMockBuilder(UuidFactoryInterface::class)
-            ->getMock();
-
-        $uuidFactory->expects($this->any())
-            ->method('uuid4')
-            ->willReturnCallback(function () {
-                $uuid = static::GENERATED_UUID[$this->uuidGeneratorInvokationCount++];
-
-                return $this->originalUuidFactory->fromString($uuid);
-            });
-
-        Uuid::setFactory($uuidFactory);
-    }
-
-    protected function tearDown(): void
-    {
-        Uuid::setFactory($this->originalUuidFactory);
     }
 
     public function testEventIsGeneratedWithUniqueIdentifier(): void
     {
         $event1 = new Event();
         $event2 = new Event();
 
-        $this->assertEquals(str_replace('-', '', static::GENERATED_UUID[0]), $event1->getId());
-        $this->assertEquals(str_replace('-', '', static::GENERATED_UUID[1]), $event2->getId());
+        $this->assertRegExp('/^[a-z0-9]{32}$/', $event1->getId());
+        $this->assertRegExp('/^[a-z0-9]{32}$/', $event2->getId());
+        $this->assertNotEquals($event1->getId(), $event2->getId());
     }
 
     public function testToArray(): void
     {
+        $event = new Event();
+
         $expected = [
-            'event_id' => str_replace('-', '', static::GENERATED_UUID[0]),
+            'event_id' => $event->getId(),
```

Comment:
That's the drawback. We can't predict the UUID, we just know that it's a random UUID (see `testEventIsGeneratedWithUniqueIdentifier`).

Reply:
I personally think this is acceptable to prevent having to use the Ramsey dependency just for validating the ID is generated from there in testing.

Checking if there is an ID with an expected length and contents is fine for me.